modules/dlm/src/internalClusterTest/java/org/elasticsearch/dlm/DataLifecycleServiceIT.java

```diff
@@ -246,6 +259,137 @@ public void testUpdatingLifecycleAppliesToAllBackingIndices() throws Exception {
+            for (int i = 0; i < randomIntBetween(10, 50); i++) {
+                indexDocs(dataStreamName, randomIntBetween(1, 300));
+                // Make sure the segments get written:
+                FlushResponse flushResponse = client().admin().indices().flush(new FlushRequest(toBeRolledOverIndex)).actionGet();
+                assertThat(flushResponse.getStatus(), equalTo(RestStatus.OK));
+            }
+            /*
+             * Without the following, calls to forcemerge are essentially a no-op since it has already done automatic merging. Setting
+             * merge_factor on its own does not do anything, but it results in calls to forcemerge making observable changes to the
+             * number of segments. So we're doing this just so that we can check that DLM did actually call forcemerge.
+             */
+            updateIndexSettings(
+                Settings.builder().put(MergePolicyConfig.INDEX_MERGE_POLICY_MERGE_FACTOR_SETTING.getKey(), 5),
+                toBeRolledOverIndex
+            );
+
+            final String toBeForceMergedIndex;
+            final int preDlmSegmentsForceMergedIndex;
+            if (currentGeneration == 1) {
+                toBeForceMergedIndex = null; // Not going to be used
+                preDlmSegmentsForceMergedIndex = -1; // Not going to be used
+            } else {
+                toBeForceMergedIndex = DataStream.getDefaultBackingIndexName(dataStreamName, currentGeneration - 1);
+                preDlmSegmentsForceMergedIndex = getSegmentCount(toBeForceMergedIndex);
+            }
+            final int preDlmSegmentsAboutToBeRolledOverIndex = getSegmentCount(toBeRolledOverIndex);
+            /*
+             * Here we briefly enable DLM so that it will run exactly once. We expect for a new write index to get created and for the one
+             * stored in toBeRolledOverIndex to get rolled over. And the one stored in toBeForceMergedIndex (if we're in the
+             * 2nd generation or later and it exists) to get force merged.
+             */
+            TimeValue dlmPollInterval = TimeValue.timeValueMillis(1000);
+            enableDLM(dlmPollInterval);
+            long dlmEnabledTime = System.nanoTime();
+            int currentBackingIndexCount = currentGeneration;
+            assertBusy(() -> {
+                if (TimeValue.timeValueNanos(System.nanoTime()).millis() - TimeValue.timeValueNanos(dlmEnabledTime)
+                    .millis() > dlmPollInterval.millis()) {
+                    /*
+                     * We want to disable DLM immediately after it has run the first time so that it doesn't roll over the latest
+                     * generation or force merge the second one.
+                     */
+                    disableDLM();
+                } else {
+                    throw new AssertionError("DLM hasn't run yet so no point in going on");
+                }
```

Comment:
I do wonder if this time math would yield some test flakiness?

Could we keep DLM disabled throughout the test and manually run iterations instead? 

e.g.
```
  DataLifecycleService dataLifecycleService = internalCluster().getInstance(DataLifecycleService.class, internalCluster().getMasterName());
  ClusterService clusterService = internalCluster().getInstance(ClusterService.class, internalCluster().getMasterName());
     
  // run DLM one
  dataLifecycleService.run(clusterService.state());
```